hikari/api/rest.py

```diff
@@ -8675,3 +8676,259 @@ async def end_poll(
+            The created auto-moderation rule.
+
+        Raises
+        ------
+        hikari.errors.BadRequestError
+            If any of the fields that are passed have an invalid value.
+        hikari.errors.ForbiddenError
+            If you are missing the `MANAGE_GUILD` permission or if you try to
+            set a TIMEOUT action without the `MODERATE_MEMBERS` permission.
+        hikari.errors.UnauthorizedError
+            If you are unauthorized to make the request (invalid/missing token).
+        hikari.errors.NotFoundError
+            If the guild was not found.
+        hikari.errors.RateLimitTooLongError
+            Raised in the event that a rate limit occurs that is
+            longer than `max_rate_limit` when making a request.
+        hikari.errors.InternalServerError
+            If an internal error occurs on Discord while handling the request.
+        """
+
+    @abc.abstractmethod
+    async def edit_auto_mod_rule(
+        self,
+        guild: snowflakes.SnowflakeishOr[guilds.PartialGuild],
+        rule: snowflakes.SnowflakeishOr[auto_mod.AutoModRule],
+        /,
+        name: undefined.UndefinedOr[str] = undefined.UNDEFINED,
+        event_type: undefined.UndefinedOr[auto_mod.AutoModEventType | int] = undefined.UNDEFINED,
+        trigger: undefined.UndefinedOr[special_endpoints.AutoModTriggerBuilder] = undefined.UNDEFINED,
+        actions: undefined.UndefinedOr[typing.Sequence[special_endpoints.AutoModActionBuilder]] = undefined.UNDEFINED,
+        enabled: undefined.UndefinedOr[bool] = undefined.UNDEFINED,
+        exempt_roles: undefined.UndefinedOr[snowflakes.SnowflakeishSequence[guilds.PartialRole]] = undefined.UNDEFINED,
+        exempt_channels: undefined.UndefinedOr[
+            snowflakes.SnowflakeishSequence[channels_.PartialChannel]
+        ] = undefined.UNDEFINED,
+        reason: undefined.UndefinedOr[str] = undefined.UNDEFINED,
+    ) -> auto_mod.AutoModRule:
+        """Edit an auto-moderation rule.
+
+        Parameters
+        ----------
+        guild
+            Object or ID of the guild to edit an auto-moderation rule in.
+        rule
+            Object or ID of the auto-moderation rule to edit.
+        name
+            If specified, the rule's new name.
+        event_type
+            The type of user content creation event this rule should trigger on.
+        trigger
+            The trigger builder to edit the trigger from.
+        actions
+            If specified, a sequence of the actions to execute when this rule
+            is triggered.
+        allow_list
+            If specified, a sequence of filters which will be exempt from
+            triggering the preset trigger.
+            This supports a wildcard matching strategy which is documented at
+            https://discord.com/developers/docs/resources/auto-moderation#auto-moderation-rule-object-keyword-matching-strategies.
+            This can only be set for KEYWORD_PRESET triggers.
```

Comment:
```suggestion
        allow_list
            If specified, a sequence of filters which will be exempt from
            triggering the preset trigger.

            This supports a wildcard matching strategy which is documented at
            <https://discord.com/developers/docs/resources/auto-moderation#auto-moderation-rule-object-keyword-matching-strategies>.

            This can only be set for `KEYWORD_PRESET` triggers.
```